src/cryptography/x509/base.py

```diff
@@ -466,6 +470,10 @@ def not_valid_after(self, time):
             raise TypeError('Expecting datetime object.')
         if self._not_valid_after is not None:
             raise ValueError('The not valid after may only be set once.')
+        if time.tzinfo is not None:
+            offset = time.utcoffset()
+            offset = offset if offset else datetime.timedelta()
+            time = time.replace(tzinfo=None) - offset
```

Comment:
Please factor this out into a standalone function, and use it in `RevokedCertificateBuilder` as well